betterproto/__init__.py

```diff
@@ -435,14 +434,22 @@ def parse_fields(value: bytes) -> Generator[ParsedField, None, None]:
 
 
 class ProtoClassMetadata:
-    cls: Type["Message"]
+    oneof_group_by_field: Dict[str, str]
```

Comment:
Probably worthwhile declaring slots here:

```python
__slots__ = (
    'cls', 
    'oneof_group_by_field', 
    'oneof_field_by_group', 
    'default_gen', 
    'cls_by_field', 
    'field_name_by_number', 
    'meta_by_field_name', 
)
```
I'd be curious if you can measure a different in performance by doing so.

Reply:
Not really. The measure is to execute the following command 3 times:

```shell
$ pytest --repeat 3000 betterproto/tests/test_input.py
```

Before `__slot__` magic, it took 21.35s, 21.59s, 21.35s.
After, it took 21.70s, 21.62s, 21.52s.

According to https://book.pythontips.com/en/latest/__slots__magic.html, the purpose of `__slot__` is to save memory from a large number of objects.